crates/context/src/lib.rs

```diff
@@ -17,46 +17,158 @@ use tracing::{error, info};
 pub mod config;
 
 #[derive(Clone)]
-pub struct ApplicationManager {
+pub struct ContextManager {
+    pub config: config::ApplicationConfig,
+    pub store: calimero_store::Store,
     pub network_client: NetworkClient,
-    pub application_dir: Utf8PathBuf,
 }
 
-pub async fn start_manager(
-    config: &config::ApplicationConfig,
-    network_client: NetworkClient,
-) -> eyre::Result<ApplicationManager> {
-    let application_manager = ApplicationManager::new(network_client, config.dir.clone());
+impl ContextManager {
+    pub async fn start(
+        config: &config::ApplicationConfig,
+        store: calimero_store::Store,
+        network_client: NetworkClient,
+    ) -> eyre::Result<Self> {
+        let this = ContextManager {
+            config: config.clone(),
+            store,
+            network_client,
+        };
 
-    application_manager.boot_installed_apps().await?;
+        this.boot().await?;
 
-    Ok(application_manager)
-}
+        Ok(this)
+    }
 
-impl ApplicationManager {
-    pub fn new(network_client: NetworkClient, application_dir: Utf8PathBuf) -> Self {
-        Self {
-            network_client,
-            application_dir,
+    pub async fn add_context(
+        &self,
+        context: calimero_primitives::context::Context,
+    ) -> eyre::Result<()> {
+        // todo! ensure application is installed
+
+        let mut handle = self.store.handle();
+
+        handle.put(
+            &calimero_store::key::ContextMeta::new(context.id),
+            &calimero_store::types::ContextMeta {
+                application_id: context.application_id.0.into(),
+                last_transaction_hash: [0; 32],
```

Comment:
aside it being compact, `Hash` right now is more costly memory-wise than `[0; 32]`

but fair enough, since you introduced the type alias, I can rewrite this